mopidy_local_images/library.py

```diff
@@ -228,3 +219,6 @@ def _scan(self, uri):
         logger.debug('Scanning %s for images', uri)
         data = self.scanner.scan(uri)
         return data.get('tags', {})
+
+    def get_data_dir(self, config):
+        raise NotImplementedError  # overridden by Extension
```

Comment:
That's also a good idea.

Another thing: the function here should be private so it is not exposed to other Pykka actors.

``` python
    def _get_data_dir(self, config):
        self._migrate_old_data_dir(config)
        return Extension().get_data_dir(config)

    def _migrate_old_data_dir(self, config):
        # TODO: Remove this method when we're confident most users
        # have upgraded away from Mopidy 1.0.
        new_dir = Extension().get_data_dir(config)
        old_dir = os.path.join(config['core']['data_dir'], b'local', b'images')
        if os.path.isdir(old_dir):
            logger.info('Migrating Mopidy-Local-SQLite to new data dir')
            for filename in os.listdir(old_dir):
                old_path = os.path.join(old_dir, filename)
                new_path = os.path.join(new_dir, filename)
                logger.info('Moving %r to %r', old_path, new_path)
                os.rename(old_path, new_path)
            os.rmdir(old_dir)
```